### PR TITLE
Generate ed25519 keypair from the seed

### PR DESCRIPTION
**Pull request by klochowicz**
Pass it trough to the actor systems.

Unfortunately, it turned out to be impossible to derive peer id from maker id,
so we will need an additional CLI option on the taker side.
After this gets merged, we'll be able to easily add a default argument for the
taker based on maker logs.

### CODE REVIEW

File: maker/src/main.rs [hunk 1]
```diff
@@ -191,11 +191,13 @@ async fn main() -> Result<()> {
     let auth_username = rocket_basicauth::Username("itchysats");
     let auth_password = seed.derive_auth_password::<rocket_basicauth::Password>();
 
-    let (identity_pk, identity_sk) = seed.derive_identity();
+    let keypair = seed.derive_identities();
```
Comment (da-kami) [inline maker/src/main.rs]: ❓Why was this renamed to `keypair`? I feel identities still expresses better what this is.

File: maker/src/main.rs [hunk 2]
```diff
@@ -191,11 +191,13 @@ async fn main() -> Result<()> {
...
 
-    let hex_pk = hex::encode(identity_pk.to_bytes());
+    let peer_id = keypair.libp2p.public().to_peer_id();
```
Comment (da-kami) [inline maker/src/main.rs]: 🙃 Could also hide the way we access the internals and just have `identities.peer_id()` here.

**Review by da-kami — APPROVED** [review]

File: daemon/src/seed.rs [hunk 3]
```diff
@@ -43,6 +56,34 @@ pub trait Seed {
...
+    fn derive_ed25519_keypair(&self) -> Result<ed25519::Keypair> {
+        let mut secret = [0u8; 32];
+
+        Hkdf::<Sha256>::new(None, &self.seed())
+            .expand(b"LIBP2P_IDENTITY", &mut secret)
+            .expect("okm array is of correct length");
+
+        let keypair = ed25519::Keypair::from(
+            ed25519::SecretKey::from_bytes(secret)
+                .context("Failed to create secret key from random bytes")?,
```
Comment (luckysori) [inline daemon/src/seed.rs]: 🔧 I think this can be an `expect`. 

`SecretKey::from_bytes` only fails if we don't pass exactly 32 bytes, and the hash generated by `SHA256` is 32 bytes.

File: maker/src/main.rs (hunk 1 above)
Comment (luckysori) [reply maker/src/main.rs]: It seems like destructuring would make sense here (but yes, "keypair" isn't quite right).

File: daemon/src/seed.rs [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
 use bdk::bitcoin::Network;
 use hkdf::Hkdf;
+use libp2p_core::identity::ed25519;
+use libp2p_core::identity::Keypair;
 use rand::Rng;
 use sha2::Sha256;
 use std::convert::TryInto;
 use std::path::Path;
 
+/// Struct containing secrets for both legacy and libp2p connections.
```
Comment (luckysori) [inline daemon/src/seed.rs]: 🙃 
```suggestion
/// Struct containing keys for both legacy and libp2p connections.
```

**Review by luckysori — APPROVED** [review]

File: maker/src/main.rs (hunk 2 above)
Comment (klochowicz) [reply maker/src/main.rs]: great suggestion!

File: maker/src/main.rs (hunk 1 above)
Comment (klochowicz) [reply maker/src/main.rs]: good catch! was meant to be renamed to identities indeed. Unfortunately, can't destructure here as it's being passed to MakerActorSystem

File: daemon/src/seed.rs (hunk 3 above)
Comment (klochowicz) [reply daemon/src/seed.rs]: sure!
